project-base/storefront/components/Pages/ProductDetail/ProductDetailContent.tsx

```diff
@@ -40,9 +41,12 @@ export const ProductDetailContent: FC<ProductDetailContentProps> = ({ product, f
     const { isProductInComparison, toggleProductInComparison, isPopupCompareOpen, setIsPopupCompareOpen } =
         useComparison();
     const { toggleProductInWishlist, isProductInWishlist } = useWishlist();
+    const { setLastVisitedProductCatalogNumbers } = useLastVisited();
```

Comment:
Why are we using this hook here and not in root of the page (`[productSlug].tsx`)? So it can handle also main variant.

Reply:
Maybe I got it. Last visited setter moved to productSlug, so it will set cookies for all product types.